crates/rolldown/src/ecmascript/ecma_generator.rs

```diff
@@ -91,7 +92,15 @@ impl Generator for EcmaGenerator {
 
         // TODO indent chunk content for iife format
         if matches!(ctx.options.format, OutputFormat::Iife) {
-          concat_source.add_source(Box::new(RawSource::new("(function() {\n".to_string())));
+          let has_exports = !get_export_items(ctx.chunk, ctx.link_output).is_empty();
+          if let Some(name) = &ctx.options.name {
+            concat_source.add_source(Box::new(RawSource::new(format!(
+              "var {name} = (function({}) {{\n",
+              if has_exports { "exports" } else { "" }
```

Comment:
Probably should think about if we follows this behavior? Or we don't introduce special behavior for single default export.

- https://rollupjs.org/repl/?version=4.18.1&shareable=JTdCJTIyZXhhbXBsZSUyMiUzQSUyMiUyMiUyQyUyMm1vZHVsZXMlMjIlM0ElNUIlN0IlMjJjb2RlJTIyJTNBJTIyZXhwb3J0JTIwZGVmYXVsdCUyMDQyJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlMkMlMjJuYW1lJTIyJTNBJTIybWFpbi5qcyUyMiU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJvdXRwdXQlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJpaWZlJTIyJTJDJTIybmFtZSUyMiUzQSUyMkElMjIlN0QlN0QlN0Q=
- https://rollupjs.org/repl/?version=4.18.1&shareable=JTdCJTIyZXhhbXBsZSUyMiUzQSUyMiUyMiUyQyUyMm1vZHVsZXMlMjIlM0ElNUIlN0IlMjJjb2RlJTIyJTNBJTIyZXhwb3J0JTIwY29uc3QlMjBhJTIwJTNEJTIwMSU1Q25leHBvcnQlMjBkZWZhdWx0JTIwNDIlMjIlMkMlMjJpc0VudHJ5JTIyJTNBdHJ1ZSUyQyUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTdEJTVEJTJDJTIyb3B0aW9ucyUyMiUzQSU3QiUyMm91dHB1dCUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmlpZmUlMjIlMkMlMjJuYW1lJTIyJTNBJTIyQSUyMiU3RCU3RCU3RA==

Reply:
> Here has some complex behavior we need to aligin it. like https://github.com/rollup/rollup/blob/master/src/finalisers/shared/getExportBlock.ts#L224. We also only add one at cjs format at now.

This goes to https://github.com/rolldown/rolldown/issues/1628. Should have an discussion first to see if we truely need to support it.